src/main/java/ch/cern/eam/wshub/core/services/material/impl/PartMiscServiceImpl.java

```diff
@@ -237,6 +239,15 @@ public String createIssueReturnTransaction(InforContext context, IssueReturnPart
 
 			issueReturnTransactionInfor.getIssueReturnTransactionLines().getIssueReturnTransactionLine()
 					.add(issueReturnTransactionLine);
+
+			if (isNotEmpty(line.getAssetIDCode())
+					&& issueReturnPartTransaction.getTransactionType().equalsIgnoreCase("ISSUE")
+					&& issueReturnPartTransaction.getTransactionOn().equals(IssueReturnPartTransactionType.WORKORDER)) {
```

Comment:
Always safer to call the equals method on the Object we know not to be null

Reply:
true, I have tested and pushed those changes, please have a look